src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -131,14 +131,14 @@ pub trait StorageController: Debug + Send {
 
 /// Controller state maintained for each storage instance.
 #[derive(Debug)]
-pub struct StorageControllerState<T> {
+pub struct StorageControllerState<T, S = mz_stash::Sqlite> {
```

Comment:
Can a rust person tell me if this is ok or there's a better way? I tried to make `StorageControllerState` generic over `S` but then `StorageControllerState::new` complained that it was always producing a `Sqlite`, not an `S`. I then tried to put Box and dyn in various places unsuccessfully. I ended up with this because it compiled?

Reply:
I know we went with this, but for posterity, you can move the `new` method into an `impl StorageControllerState<Sqlite>` block, and leave everything else in the `impl< S: Stash> StorageControllerState<S>` block